spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerFactsSpecTest.java

```diff
@@ -0,0 +1,195 @@
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+        OffsetDateTime updatedDate = updatedConsumer.getUpdated();
+
+        assertThat(updatedDate)
+            .isAfter(initialDate);
+    }
+
+    @Test
+    public void shouldAllowRemovingAllFacts() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.getFacts().clear();
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .isEmpty();
+    }
+
+    @Test
+    @SuppressWarnings("indentation")
+    public void shouldAllowFactKeysAndValuesOfLength255() {
+        String key = "a".repeat(255);
+        String value = "b".repeat(255);
+        ConsumerDTO consumer = ownerClient.consumers()
+            .createConsumer(Consumers.random(this.owner)
+                .putFactsItem(key, value)
+            );
+
+        assertThat(consumer.getFacts())
+            .containsEntry(key, value);
+    }
+
+    @Test
+    public void shouldTruncateLongFactKeys() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.putFactsItem("a".repeat(256), "some_value");
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .containsKey("a".repeat(252) + "...");
+    }
+
+    @Test
+    public void shouldTruncateLongFactValues() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.putFactsItem(Facts.UnameMachine.key(), "a".repeat(256));
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .containsEntry(Facts.UnameMachine.key(), "a".repeat(252) + "...");
+    }
+
+    private void sleepForMs(int millis) {
+        try {
+            Thread.sleep(millis);
+        }
+        catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SuppressWarnings("indentation")
+    private ConsumerDTO createConsumer() {
+        return ownerClient.consumers()
+            .createConsumer(Consumers.random(this.owner)
+                .putFactsItem(Facts.UnameMachine.key(), "i686")
+                .putFactsItem(Facts.UnameSystem.key(), "Linux")
+            )
+            .addInstalledProductsItem(
```

Comment:
This bit should probably be included in the part that's sent up to the server, so this util method isn't returning an already-modified consumer DTO.

In general it's kind of rough to read due to the indentation+chaining shenanigans.

Reply:
Good catch. I shuffled the chain a bit to remove the indentation.